### DISCUSSION TITLE
Same traceId when a batch is consumed from kafka

### DISCUSSION THREAD

**Discussion by shravani154** [open-telemetry/opentelemetry-java-instrumentation#6094] [Q&A]
In java Instrumentation when a batch is consumed from kafka/eventhub the entire batch creates a single trace, thus all messages in the batch have the same trace id. Even when the batch-mode property is set to false I see same trace id for all the messages. I want to have a different trace id for all the messages consumed. I tried setting max-poll-records consumer property to 1, then it solves my usecase giving different trace ids but it is affecting the performance when data sent in larger volume. Any solution which solves my use case is appreciable.

**Comment by trask** [comment]
hi @shravani154! which OpenTelemetry Javaagent version are you using? and how are you consuming the batch?

**Reply by shravani154** [reply]
I am using v1.11.0. I am using spring cloud streams kafka where I have a consumer bean to consume messages.

**Reply by trask** [reply]
can you try with 1.14.0 just to confirm if there's any different behavior? there have been several improvements to kafka messaging instrumentation in recent versions
